hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -906,4 +987,23 @@ private Map<String, String> supplementOptions(
       return newOptions;
     }
   }
+
+  private HoodieFlinkWriteClient<?> createWriteClient(
+      String tablePathStr,
+      ObjectPath tablePath) throws Exception {
+    Map<String, String> options = supplementOptions(tablePath, translateSparkTable2Flink(tablePath, getHiveTable(tablePath)).getParameters());
+    // enable auto-commit though ~
```

Comment:
Just invoke the #getTable to fetch the catalog table first, we can then use the table options.